src/snowflake/snowpark/dataframe.py

```diff
@@ -3336,82 +3346,100 @@ def _show_string(self, n: int = 10, max_width: int = 50, **kwargs) -> str:
         if self._session._conn.is_phase1_enabled():
             ast = self._session._ast_batch.flush()
             res = self._session._conn.ast_query(ast)
-            print(f"AST response: {res}")
-        else:
-            _, kwargs["_dataframe_ast"] = self._session._ast_batch.flush()
 
-        if is_sql_select_statement(query):
-            result, meta = self._session._conn.get_result_and_metadata(
-                self.limit(n)._plan, **kwargs
+            _logger.debug(f"AST response: {res}")
+
+            # In Phase 1, the code to format the result set to a string
+            # is run on the server, retrieve simply the result here.
+            response = decode_ast_response_from_snowpark(
+                res, self._session._conn._conn._session_parameters
             )
+
+            check_response(response)
+
+            return response.body[0].eval_ok.data.string_val.v
         else:
-            res, meta = self._session._conn.get_result_and_metadata(
```

Comment:
this code has been moved under the else. It is only relevant for phase0 with the result path added here for phase1.